geode-membership/src/main/java/org/apache/geode/distributed/internal/membership/gms/MemberIdentifierImpl.java

```diff
@@ -284,7 +279,7 @@ public int compareTo(MemberIdentifier other, boolean compareMemberData,
 
     String myName = getName();
     String otherName = other.getName();
-    if (!(other.isPartial() || this.isPartial())) {
+    if (!(other.isPartial() || isPartial())) {
       if (myName == null && otherName == null) {
```

Comment:
The warning about an empty if statement here can be fixed by refactoring the block to:
```
      if (myName != null || otherName != null) {
        if (myName == null) {
          return -1;
        } else if (otherName == null) {
          return 1;
        } else {
          int i = myName.compareTo(otherName);
          if (i != 0) {
            return i;
          }
        }
      }  
```

Reply:
I want to. I really do. I was worried about the readability change in similar empty blocks like this. Let me noodle it and see if refactoring it changes the readability.

Reply:
A bit of a rabbit hole but I like it better now.

Reply:
Oh yeah, that's waaaay better now, nice.

Reply:
I really wish we were at least Java 9 already. There are array comparators in 9. Some day... hopefully before I retire.